vividus-analytics/src/main/java/org/vividus/analytics/AnalyticsStoryReporter.java

```diff
@@ -108,16 +117,14 @@ public void beforeStep(Step step)
         STEPS.incrementAndGet();
     }
 
-    private void postPluginsAnalytic(Map<String, String> modules)
+    private List<AnalyticsEvent> collectPluginsAnalytic(Map<String, String> modules)
     {
-        modules.forEach((k, v) -> {
-            if (k.startsWith("vividus-plugin-"))
-            {
-                Map<String, String> payload = new HashMap<>();
-                CustomDefinitions.PLUGIN_VERSION.add(payload, v);
-                eventBus.post(new AnalyticsEvent(k, "use", payload));
-            }
-        });
+        return modules.entrySet().stream().filter(module -> module.getKey().startsWith("vividus-plugin-")).map(module ->
```

Comment:
What about the agents?

Reply:
Looks like a new feature request :) the goal of this PR is optimisation of I/O operations